bus/routes.go

```diff
@@ -484,6 +484,30 @@ func (b *Bus) walletPendingHandler(jc jape.Context) {
 	jc.Encode(events)
 }
 
+func (b *Bus) hostsHandlerGET(jc jape.Context) {
+	var offset int
+	if jc.DecodeForm("offset", &offset) != nil {
+		return
+	} else if offset < 0 {
+		jc.Error(api.ErrInvalidOffset, http.StatusBadRequest)
+		return
+	}
+
+	limit := -1
+	if jc.DecodeForm("limit", &limit) != nil {
+		return
+	} else if limit < -1 {
+		jc.Error(api.ErrInvalidLimit, http.StatusBadRequest)
+		return
+	}
+
+	hosts, err := b.store.UsableHosts(jc.Request.Context(), offset, limit)
+	if jc.Check("couldn't fetch hosts", err) != nil {
+		return
+	}
+	jc.Encode(hosts)
```

Comment:
I think we need to perform gouging checks here before returning the hosts. The autopilot doesn't perform gouging checks frequently enough.
Otherwise we might return a "good" host here and then immediately fail to fund it because the funding endpoint performance live gouging checks.

Reply:
That messes up `offset` and `limit`, perhaps we can perform the check before populating the cache 🤔 .

Reply:
The worker shouldn't need to perform any gouging checks on the returned hosts. At least once we got rid of contracts completely.

I also think it's fine to only apply `offset` and `limit` within the handler of the endpoint or to drop it completely. We won't use it anytime soon and the response is tiny. We'll eventually also just keep that list in memory.